### PR TITLE
Email notifications for moderation

### PR DESCRIPTION
**Pull request by deiu**
Added email notifications for moderation.

## Description

This PR implements a helper function that uses SendGrid to send of emails. In this instance we are using it to send email notifications every time a new moderation request is made (i.e. a user has reported content). Emails are sent **only once** per content identifier that was reported -- i.e. a second user reporting the same content will **NOT** trigger a new email to be sent.

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

**Comment by deiu** [comment]
Ready for review.

File: api/hub/src/datasources/moderation-report.ts [hunk 1]
```diff
@@ -195,6 +195,26 @@ class ModerationReportAPI extends DataSource {
...
+      const text = `There is a new pending request for moderation.
+      Please visit the moderation app (https://akasha.ethereum.world/moderation-app/home)
+      to moderate this content.
+      \nThank you!`;
+      const html = `<p>There is a new pending request for moderation.
+      Please visit the <a href="https://akasha.ethereum.world/moderation-app/home">moderation app</a>
+      to moderate this content.</p>
+      <br>
+      <p>Thank you!</p>`;
+      try {
```
Comment (kenshyx) [inline api/hub/src/datasources/moderation-report.ts]: this should have a debounce period like 1h, or we end up with sending emails for every report made

File: api/hub/src/datasources/moderation-report.ts [hunk 2]
```diff
@@ -195,6 +195,26 @@ class ModerationReportAPI extends DataSource {
...
+        await sendEmailNotification(process.env.MODERATION_EMAIL,
```
Comment (kenshyx) [inline api/hub/src/datasources/moderation-report.ts]: if you need to try catch then you can use directly the promise api `sendEmailNotification(...).then(_=>whatever).catch(err => do err stuff)`

File: api/hub/src/helpers.ts [hunk 3]
```diff
@@ -209,6 +210,32 @@ export const sendAuthorNotification = async (
...
+ * @returns A promise that resolves upon sending the email
+ */
+export const sendEmailNotification = async (
+to: string,
+from: string,
+subject: string,
+text: string,
+html: string
+) => {
+  sendgrid.setApiKey(process.env.SENDGRID_API_KEY)
```
Comment (kenshyx) [inline api/hub/src/helpers.ts]: you dont need to set this on every send call, set it up outside

File: api/hub/src/helpers.ts [hunk 4]
```diff
@@ -209,6 +210,32 @@ export const sendAuthorNotification = async (
...
+ * Send an email notification using SendGrid.
+ * @param to - Destination email address
+ * @param from - Source email address
+ * @param subject - Email subject
+ * @param text - Email body (text version)
+ * @param html - Email body (HTML version)
+ * @returns A promise that resolves upon sending the email
+ */
+export const sendEmailNotification = async (
+to: string,
```
Comment (kenshyx) [inline api/hub/src/helpers.ts]: consider using an object param with these attributes, makes the function signature easier to read when it is called

File: api/hub/src/datasources/moderation-report.ts (hunk 1 above)
Comment (deiu) [reply api/hub/src/datasources/moderation-report.ts]: Actually, this email is only sent on the first report for each content ID (when the pending moderation item is created). All subsequent reports will not trigger it.

File: api/hub/src/datasources/moderation-report.ts (hunk 2 above)
Comment (deiu) [reply api/hub/src/datasources/moderation-report.ts]: Done.

**Review by kenshyx — APPROVED** [review]

**Pull request merged**